packages/main/src/Table.js

```diff
@@ -574,6 +586,39 @@ class Table extends UI5Element {
 		}
 	}
 
+	_handleTab(event) {
+		const target = getNormalizedTarget(event.target);
+		const isNext = isTabNext(event);
+		const isHeaderFocused = target === this.getColumnHeader();
+		const isMoreBtnFocused = target === this.getMoreButton();
+
+		if (isNext) {
+			if (isHeaderFocused) {
+				if (!this.growsWithButton) {
+					this._focusForwardElement(event, true);
+					this.lastFocusedElement = this.getColumnHeader();
+				} else {
+					this.getMoreButton().focus();
+				}
+			} else if (isMoreBtnFocused) {
+				this._focusForwardElement(event, true);
+			}
+
+			return;
+		}
+
+		if (isHeaderFocused) {
+			this._focusForwardElement(event, false);
+			this.lastFocusedElement = this.getColumnHeader();
+		} else if (isMoreBtnFocused) {
+			if (this.lastFocusedElement) {
+				this.lastFocusedElement.focus();
+			} else {
+				this.currentElement.focus();
+			}
+		}
+	}
```

Comment:
This many nested if/ if/else statements are not really easy to follow. May be you can try using switch instead and have it executed from the focused element's perspective (header, grow button, row, etc). This way there should be less cases for each element

Reply:
I isolated part of the logic in `switch`es, but not all of it as there are many specific cases and the simple nature of a switch does not suffice.